File: Problem List/13. Roman to Integer.py
# Best Big O
class Solution:
    def romanToInt(self, s: str) -> int:
        values = {'I': 1, 'V': 5, 'X': 10, 'L': 50, 'C': 100, 'D': 500, 'M': 1000}
        i = len(s)-1
        sum_ = values[s[i]]
        while i > 0:
            i -=1
            if values[s[i]] < values[s[i+1]]:
                sum_ -= values[s[i]]
            else:
                sum_ += values[s[i]]
        return sum_

File: Problem List/13. Roman to Integer.c
int romanCharToInt(char c)
{
  switch (c)
  {
  case 'I': return 1;
  case 'V': return 5;
  case 'X': return 10;
  case 'L': return 50;
  case 'C': return 100;
  case 'D': return 500;
  case 'M': return 1000;
  }
  return 0;
}

int romanToInt(char *s)
{
  int i = 0, sum = 0;
  while (s[i] != '\0')
  {
    int value1 = romanCharToInt(s[i]);
    int value2 = romanCharToInt(s[i + 1]);

    if (value2 > value1)
    {
      sum += value2 - value1;
      i += 2;
    }
    else
    {
      sum += value1;
      i++;
    }
  }
  return sum;
}
